src/server/server_family.cc

```diff
@@ -3470,6 +3505,9 @@ void ServerFamily::AddReplicaOf(CmdArgList args, const CommandContext& cmd_cntx)
   }
   add_replica->StartMainReplicationFiber(nullopt);
   cluster_replicas_.push_back(std::move(add_replica));
+  service_.proactor_pool().AwaitFiberOnAll(
+      [this](auto index, auto* cntx)
```

Comment:
this is not the right way to do it. `cluster_replicas_` can change in between. 
instead you should capture it in the callback by value instead of `this`.